x-pack/plugins/canvas/server/usage/workpad_collector.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { sum as arraySum, min as arrayMin, max as arrayMax, get } from 'lodash';
+import { fromExpression } from '@kbn/interpreter/common';
+import { CANVAS_TYPE } from '../../common/lib/constants';
+import { collectFns } from './collector_helpers';
+
+export function handleResponse({ hits }) {
+  const workpadDocs = get(hits, 'hits', null);
+  if (workpadDocs == null) {
+    return;
+  }
+
+  const functionSet = new Set();
+
+  // make a summary of info about each workpad
+  const workpadsInfo = workpadDocs.map(hit => {
+    const workpad = hit._source[CANVAS_TYPE];
+
+    let pages;
+    try {
+      pages = { count: workpad.pages.length };
+    } catch (err) {
+      console.warn(err, workpad);
+    }
+    const elementCounts = workpad.pages.reduce(
+      (accum, page) => accum.concat(page.elements.length),
+      []
+    );
+    const functionCounts = workpad.pages.reduce((accum, page) => {
+      return page.elements.map(element => {
+        const ast = fromExpression(element.expression);
+        collectFns(ast, cFunction => {
+          functionSet.add(cFunction);
+        });
+        return ast.chain.length; // get the number of parts in the expression
+      });
+    }, []);
+    return { pages, elementCounts, functionCounts };
+  });
+
+  // combine together info from across the workpads
+  const combinedWorkpadsInfo = workpadsInfo.reduce(
+    (accum, pageInfo) => {
+      const { pages, elementCounts, functionCounts } = pageInfo;
+
+      return {
+        pageMin: pages.count < accum.pageMin ? pages.count : accum.pageMin,
+        pageMax: pages.count > accum.pageMax ? pages.count : accum.pageMax,
+        pageCounts: accum.pageCounts.concat(pages.count),
+        elementCounts: accum.elementCounts.concat(elementCounts),
+        functionCounts: accum.functionCounts.concat(functionCounts),
+      };
+    },
+    {
+      pageMin: Infinity,
+      pageMax: -Infinity,
```

Comment:
This is so odd to see... can you explain the purpose of this?  Perhaps a reference to the thinking behind it?